### DISCUSSION TITLE
What is the intuition of negative attention?

### DISCUSSION THREAD

**Discussion by wanglu2014** [pyg-team/pytorch_geometric#2471] [General]
What does negative attention mean? It doesn't matter, or is it kind of a negative correlation?

**Comment by rusty1s** [comment]
I'm not sure to which GNN operator you are referring to. Attention scores usually to fit the interval `[0, 1]`, e.g., via the softmax function.

**Reply by wanglu2014** [reply]
It is attention weight of GAT. Before softmax, the weights contain both negative and positive values.

**Reply by rusty1s** [reply]
Yes, high negative values will push the values after softmax close to zero.

**Reply by adhikarirsr** [reply]
Before Softmax `e_{ij}` can be negative as well as all positive because of `LeakyReLU`. However, as pointed out by @rusty1s using softmax it is normalized. It is explained pretty nicely here: https://www.youtube.com/watch?v=A-yKQamf2Fc&ab_channel=DeepFindr
